libraries/OpenThread/src/OThreadCLI.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include "soc/soc_caps.h"
+#include "sdkconfig.h"
+#if SOC_IEEE802154_SUPPORTED
+#if CONFIG_OPENTHREAD_ENABLED
+
+#include "esp_openthread.h"
+#include "esp_openthread_cli.h"
+#include "esp_openthread_lock.h"
+#include "esp_openthread_netif_glue.h"
+#include "esp_openthread_types.h"
+
+#include "openthread/cli.h"
+#include "openthread/instance.h"
+#include "openthread/logging.h"
+#include "openthread/tasklet.h"
+#include "openthread/dataset_ftd.h"
+
+#include "Arduino.h"
+
+typedef std::function<void(void)> OnReceiveCb_t;
+
+class OpenThreadCLI : public Stream {
+private:
+  static size_t setBuffer(xQueueHandle &queue, size_t len);
+  bool otStarted = false;
+public:
+  OpenThreadCLI();
+  ~OpenThreadCLI();
+  // returns true if OpenThread CLI is running
+  operator bool() const;
+
+  // starts a task to read/write otStream. Default prompt is "ot> ". Set it to NULL to make it invisible.
+  void startOpenThreadConsole(Stream& otStream, bool echoback = true, const char *prompt = "ot> ");
+  void stopOpenThreadConsole();
```

Comment:
how about `startConsole` and `endConsole`?

Reply:
ok.

Reply:
Done.